### PR TITLE
feat(medusa): Refund line totals calculation

### PR DESCRIPTION
**Pull request by adrien2p**
Rely on the update of the following pr https://github.com/medusajs/medusa/pull/2136

**WIP Missing integration tests**

**What**

Update the totals calculation on the refund line to include the notion of tax inclusive

**Test**

- Update and add new tests around the refund

Fixes CORE-482

### CODE REVIEW

File: integration-tests/api/__tests__/returns/ff-tax-inclusive-pricing.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    discounts,
+    line_items: [
+      {
+        id: "test-item",
+        variant_id: "test-variant",
+        quantity: 2,
+        fulfilled_quantity: options?.shipped ? 2 : undefined,
+        shipped_quantity: options?.shipped ? 2 : undefined,
+        unit_price: options?.oldTaxes ? 1125 : 1200,
+        includes_tax: true,
```
Comment (srindom) [inline integration-tests/api/__tests__/returns/ff-tax-inclusive-pricing.js:48]: suggestion(non-blocking): maybe add 1-2 tests where `includes_tax: false` to cover that as well?

File: packages/medusa/src/services/totals.ts [hunk 2]
```diff
@@ -531,6 +549,23 @@ class TotalsService extends TransactionBaseService {
...
+            price: lineItem.unit_price,
+            taxRate,
+            includesTax,
+          })
+        )
+
+    lineSubtotal =
+      (lineItem.unit_price - taxAmountIncludedInPrice) * lineItem.quantity -
+      discountAmount
+
```
Comment (srindom) [inline packages/medusa/src/services/totals.ts:568]: thought: round this value for good measure?

**Review by srindom — COMMENTED** [review]
Great work - two minor comments

File: packages/medusa/src/services/totals.ts (hunk 2 above)
Comment (adrien2p) [reply packages/medusa/src/services/totals.ts:568]: It is rounded just after applying the rate is it not enough you think?

File: integration-tests/api/__tests__/returns/ff-tax-inclusive-pricing.js (hunk 1 above)
Comment (adrien2p) [reply integration-tests/api/__tests__/returns/ff-tax-inclusive-pricing.js:48]: Yeah sure

Comment (adrien2p) [reply integration-tests/api/__tests__/returns/ff-tax-inclusive-pricing.js]: done

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**